src/extensions/surveys/surveys-utils.tsx

```diff
@@ -671,3 +678,105 @@ export const renderChildrenAsTextOrHtml = ({ component, children, renderAsHtml,
               style,
           })
 }
+
+export function getNextStep(
+    survey: Survey,
+    question: SurveyQuestion,
+    displayQuestionIndex: number,
+    response: string | string[] | number | null
+) {
+    const nextQuestionIndex = displayQuestionIndex + 1
+
+    if (!question.branching?.type) {
+        if (displayQuestionIndex === survey.questions.length - 1) {
+            return SurveyQuestionBranchingType.ConfirmationMessage
+        }
+
+        return nextQuestionIndex
+    }
+
+    if (question.branching.type === SurveyQuestionBranchingType.ConfirmationMessage) {
+        return SurveyQuestionBranchingType.ConfirmationMessage
+    } else if (question.branching.type === SurveyQuestionBranchingType.SpecificQuestion) {
+        if (Number.isInteger(question.branching.index)) {
+            return question.branching.index
+        }
+    } else if (question.branching.type === SurveyQuestionBranchingType.ResponseBased) {
+        // Single choice
+        if (question.type === SurveyQuestionType.SingleChoice) {
+            // :KLUDGE: for now, look up the choiceIndex based on the response
+            // TODO: once QuestionTypes.MultipleChoiceQuestion is refactored, pass the selected choiceIndex into this method
+            const selectedChoiceIndex = question.choices.indexOf(`${response}`)
+
+            if (question.branching?.responseValues?.hasOwnProperty(selectedChoiceIndex)) {
```

Comment:
We should make this part of the SDK like `getActiveMatchingSurveys`, to `getNextQuestion(survey, currentQuestionIndex, response) -> nextQuestionIndex` for better API mode handling.

Plus, this means we can much more flexibly change things here later, which is bound to happen imo (new branching options, changing brackets, etc. etc.) - all of which would be very hard to handle for API mode if we don't have it as an SDK level primitive.

Reply:
makes sense and done - named it `getNextSurveyStep` since it returns either `index` or `confirmation_message`